### PR TITLE
Added setting to skip new account generation. "CREATE_NEW_USER"

### PR DESCRIPTION
**Pull request by CaptainPi-PiLabs**

### CODE REVIEW

File: azure_auth/handlers.py [hunk 1]
```diff
@@ -92,6 +92,8 @@ def authenticate(self, token: dict) -> UserModel:
         try:
             user = UserModel._default_manager.get_by_natural_key(email)
         except UserModel.DoesNotExist:
+            if not settings.AZURE_AUTH.get("CREATE_NEW_USER", True):
```
Comment (regoawt) [inline azure_auth/handlers.py:95]: Will need a `self.request.session.flush()` here to clear the id_token claims from the session I reckon.

**Review by regoawt — CHANGES_REQUESTED** [review]
Could you also add a test to the `TestCallbackView` test case, I think this should be sufficient:
```
@override_settings(AZURE_AUTH={
    "CLIENT_ID": "dummy_client_id",
    "CLIENT_SECRET": "dummy_client_secret",
    "REDIRECT_URI": "http://mydomain/azure_auth/callback",
    "SCOPES": ["User.Read"],
    "AUTHORITY": "https://login.microsoftonline.com/dummy_tenant_id",
    "LOGOUT_URI": "http://mydomain/logout",
    "PUBLIC_URLS": ["public", "decorator_protected"],
    "PUBLIC_PATHS": ["/public_external"],
    "CREATE_NEW_USER": False
})
    def test_callback_do_not_create_new_user(self, mocked_msal_app, mocked_requests, mocked_cache):
        mocked_msal_app.return_value.acquire_token_by_auth_code_flow.return_value = (
            self.token
        )
        mocked_cache.has_state_changed = True
        mocked_cache.serialize = msal.SerializableTokenCache().serialize
        mocked_cache.deserialize = msal.SerializableTokenCache().deserialize

        # Generate unsaved new user instance
        custom_mixer = Mixer(commit=False)
        new_user = custom_mixer.blend(UserModel)
        assert len(UserModel.objects.all()) == 1

        # Graph API response
        expected_response_json = self._get_graph_response(new_user)
        mocked_requests.get.side_effect = [
            self._mocked_response(HTTPStatus.OK, expected_response_json)
        ]
        resp = self.client.get(reverse("azure_auth:callback"))
        assert resp.status_code == HTTPStatus.FORBIDDEN
        assert "id_token_claims" not in self.client.session

        self._msal_asserts(mocked_msal_app)
        self._graph_asserts(mocked_requests)

        # Confirm user not created
        with pytest.raises(UserModel.DoesNotExist):
            UserModel.objects.get(email=new_user.email)
```

**Comment by CaptainPi-PiLabs** [comment]
Been trying to figure out how pytest works but I'm clueless

**Comment by regoawt** [comment]
Hi, I've been thinking about the use case for this and have concluded that instead of doing access control this way, one should do it via AAD/Entra directly as per https://learn.microsoft.com/en-us/entra/identity-platform/howto-restrict-your-app-to-a-set-of-users.

**Pull request closed without merging**